lib/src/controls/flyouts/flyout.dart

```diff
@@ -0,0 +1,779 @@
+                        final parentBox =
+                            context.findAncestorRenderObjectOfType<
+                                RenderCustomSingleChildLayoutBox>()!;
+                        final delegate =
+                            parentBox.delegate as _FlyoutPositionDelegate;
+
+                        final realPlacementMode = delegate.autoPlacementMode;
+
+                        return transitionBuilder!(
+                          context,
+                          animation,
+                          realPlacementMode ?? delegate.placementMode,
+                          flyout,
+                        );
+                      }),
+                    ),
+                  ),
+                ),
+              ),
+              ...menus,
+            ]);
+
+            if (dismissOnPointerMoveAway) {
+              box = MouseRegion(
+                onHover: (hover) {
+                  if (flyoutKey.currentContext == null) return;
+
+                  // the flyout box needs to be fetched at each [onHover] because the
+                  // flyout size may change (a MenuFlyout, for example)
+                  final flyoutBox =
+                      flyoutKey.currentContext!.findRenderObject() as RenderBox;
+                  final flyoutRect =
+                      flyoutBox.localToGlobal(Offset.zero) & flyoutBox.size;
+                  final menusRects = keys.map((key) {
+                    if (key.currentContext == null) return Rect.zero;
+
+                    final menuBox =
+                        key.currentContext!.findRenderObject() as RenderBox;
+                    return menuBox.localToGlobal(Offset.zero) & menuBox.size;
+                  });
+
+                  if (!flyoutRect.contains(hover.position) &&
+                      !targetRect.contains(hover.position) &&
+                      !menusRects
+                          .any((rect) => rect.contains(hover.position))) {
+                    navigator.pop();
+                  }
+                },
+                child: box,
+              );
+            }
+
+            if (dismissWithEsc) {
+              box = Actions(
+                actions: {DismissIntent: _DismissAction(navigator.pop)},
+                child: FocusScope(
+                  autofocus: true,
+                  child: box,
+                ),
+              );
+            }
+
+            return FadeTransition(
+              opacity: CurvedAnimation(
+                curve: Curves.ease,
+                parent: animation,
+              ),
+              child: box,
+            );
+          },
+        );
+      },
+    ));
+
+    _open = false;
+    notifyListeners();
+
+    return result;
+  }
+}
+
+class _DismissAction extends DismissAction {
+  _DismissAction(this.onDismiss);
+
+  final VoidCallback onDismiss;
+
+  @override
+  void invoke(covariant DismissIntent intent) {
+    onDismiss();
+  }
+}
+
+class FlyoutAttach extends StatefulWidget {
```

Comment:
`FlyoutTarget` instead maybe? Sounds better, FlyoutAttach doesn't really tell you clearly what it's meant for